operator/v1/types_console.go

```diff
@@ -143,8 +143,133 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeMode is the value of the logo theme mode that determines the theme mode in the console UI.
+// +kubebuilder:validation:Enum="Dark";"Light"
+// +enum
+type ThemeMode string
+
+// ThemeMode values
+const (
+	// ThemeModeDark represents the dark theme for the console UI.
+	ThemeModeDark ThemeMode = "Dark"
+
+	// ThemeModeLight represents the light theme for the console UI.
+	ThemeModeLight ThemeMode = "Light"
+)
+
+// LogoType is the value of the logo type that determines if the logo is for the masthead or the favicon in the console UI.
+// The masthead logo is displayed in the masthead and about modal of the console UI.
+// +kubebuilder:validation:Enum="Masthead";"Favicon"
+// +enum
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// SourceType defines the source type of the file reference.
+// +kubebuilder:validation:Enum="ConfigMap"
+// +enum
+type SourceType string
+
+const (
+	// SourceTypeConfigMap represents a ConfigMap source.
+	SourceTypeConfigMap SourceType = "ConfigMap"
+)
+
+// ConfigMapFileReference references a specific file within a ConfigMap.
+type ConfigMapFileReference struct {
+	// name is the name of the ConfigMap.
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+
+	// key is the logo key inside the referenced ConfigMap.
+	// Must consist only of alphanumeric characters, dashes (-), underscores (_), and periods (.).
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="self.matches('.*\\\\.[a-zA-Z]{2,5}$')",message="The ConfigMap key must end with a valid file extension (2 to 5 letters)."
+	// +required
+	Key string `json:"key"`
+}
+
+type FileReferenceSource struct {
+	// from specifies the source type of the file reference.
+	// This field is an enum with a valid value "ConfigMap".
+	// +required
```

Comment:
Mention in the GoDoc that this field is required.

What happens when specifying the ConfigMap value? What additional constraints exist when specifying a specific source type?

Reply:
> Mention in the GoDoc that this field is required.
> 
> What happens when specifying the ConfigMap value? What additional constraints exist when specifying a specific source type?

I think they are stated in the [ConfigMapFileReference](https://github.com/openshift/api/pull/2177/files#diff-2d91c76ad45d9598ff5cf4b5a091f9641507a937859fca78eadcece3e6cae9e9R185) above. Earlier feedback stated that the constraints should be close to the actual field that enforces them. Should this be duplicated, or does it require additional constraints?

Reply:
I'm referring to the constraints for this field specifically. This is a discriminated union, so it should probably have constraints similar to this: https://github.com/operator-framework/operator-controller/blob/7061d92a5ab344db197c637f5afb47bd9abe6520/api/v1/clustercatalog_types.go#L227-L245